src/DocumentWatcher.ts

```diff
@@ -51,14 +51,22 @@ class DocumentWatcher implements EditorConfigProvider {
 			if (path.basename(savedDocument.fileName) === '.editorconfig') {
 				// Saved an .editorconfig file => rebuild map entirely and then
 				// apply the changes to the .editorconfig file itself
-				this._rebuildConfigMap().then(applyOnSaveTransformations.bind(
-					undefined,
-					savedDocument,
-					this
-				));
+				this._rebuildConfigMap();
+				// TODO The transformations should be applied to the `.editorconfig` file as well after the config
```

Comment:
Tried that but didn't want to work decently. Also when `insert_final_newline` is set to `true` initially and you change it to `false` and remove the final empty line, this will happen

1. Save `.editorconfig`
2. Pre-save hook executed and a newline is inserted
3. Document saved, let's save it again
4. Pre-save hook executed, we don't need to add a newline because now it is `false`
5. Final save

The `.editorconfig` will end up with an empty line. So I think we will have to add some extra logic to handle this use case.